File: Stage14/Java/Problem7785.java
package Stage14.Java;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Collections;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class Problem7785 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        int n = Integer.parseInt(br.readLine());

        Set<String> set = new HashSet<>();

        for (int i = 0; i < n; i++) {
            String[] line = br.readLine().split(" ");
            String name = line[0];
            String action = line[1];

            if (action.equals("enter")) {
                set.add(name);
            } else {
                set.remove(name);
            }
        }

        List<String> list = new ArrayList<>(set);
        Collections.sort(list, Collections.reverseOrder());

        for (String name : list) {
            sb.append(name).append("\n");
        }

        System.out.print(sb);
        br.close();
    }
}

File: Stage14/Python/Problem7785.py
def main():
    n = int(input())
    people = set()
    
    for _ in range(n):
        name, action = input().split()
        if action == 'enter':
            people.add(name)
        else:
            people.discard(name)
    
    for name in sorted(people, reverse=True):
        print(name)
    
if __name__ == "__main__":
    main()